### PR TITLE
OCPBUGS-14716: Handle branding based on API changes

### PR DESCRIPTION
**Pull request by jhadvig**
Reflecting changes from https://github.com/openshift/api/pull/1494

/assign @TheRealJon

### CODE REVIEW

File: test/e2e/brand_builtins_test.go [hunk 1]
```diff
@@ -101,14 +112,37 @@ func TestBrandingFromManagedConfigMap(t *testing.T) {
...
 		})
 		if err != nil {
 			t.Fatalf("error: brand was never updated, %v", err)
 		}
 	}
 }
 
+// Since the console-operator API contains also legacy brands, we need to test that all the brands
+// are correctly mapped to the legacy brands in the console-config.
+func compareBrands(expectedBrand, gotBrand operatorsv1.Brand) bool {
```
Comment (jhadvig) [inline test/e2e/brand_builtins_test.go:125]: @JoelSpeed

File: test/e2e/brand_builtins_test.go [hunk 2]
```diff
@@ -16,8 +16,19 @@ import (
...
+	operatorsv1.BrandOnlineLegacy,
+	operatorsv1.BrandDedicatedLegacy,
+	operatorsv1.BrandAzureLegacy,
+	operatorsv1.BrandOKD,
+	operatorsv1.BrandOCP,
+	operatorsv1.BrandOnline,
+	operatorsv1.BrandDedicated,
+	operatorsv1.BrandAzure,
+	operatorsv1.BrandROSA,
+}
```
Comment (TheRealJon) [inline test/e2e/brand_builtins_test.go:30]: nit, code style

```suggestion
var validOperatorConfigBrands = []operatorsv1.Brand{
	operatorsv1.BrandAzure,
	operatorsv1.BrandAzureLegacy,
	operatorsv1.BrandDedicated,
	operatorsv1.BrandDedicatedLegacy,
	operatorsv1.BrandOCP,
	operatorsv1.BrandOCPLegacy,
	operatorsv1.BrandOKD,
  operatorsv1.BrandOKDLegacy,
	operatorsv1.BrandOnline,
	operatorsv1.BrandOnlineLegacy,
	operatorsv1.BrandROSA,
}
```

**Review by TheRealJon — APPROVED** [review]
/lgtm

Just one nit if you choose to update that, just let me know and I'll re-tag.

**Pull request merged**